examples/all-clusters-app/esp32/README.md

```diff
@@ -120,64 +108,103 @@ If packages are already installed then simply activate it.
-    menuconfig, then ESP32 connects to the AP with those credentials (STA mode).
+Select the Rendezvous Mode via `Demo -> Rendezvous Mode`. If Rendezvous Mode is
+ByPass then set the credentials of WiFi Network i.e SSID and Password from
+menuconfig.
+
+`idf.py menuconfig -> Component config -> CHIP Device Layer -> WiFi Station Options`
 
 2.  Now flash the device with the same command as before. (Use the right `/dev`
     device)
 
           $ idf.py flash monitor ESPPORT=/dev/tty.SLAB_USBtoUART
 
-3.  The device should boot up and connect to your network. When that happens you
+3.  The device should boot up and when device connects to your network then you
     will see a log like this in the monitor.
 
           I (5524) chip[DL]: SYSTEM_EVENT_STA_GOT_IP
           I (5524) chip[DL]: IPv4 address changed on WiFi station interface: <IP_ADDRESS>...
 
-    Note: If you are using the M5Stack, the screen will display the server's IP
-    Address if it successfully connects to the configured 2.4GHz Network.
-
-4.  Use
+4.  Controller used
+    [python based device controller](https://github.com/project-chip/connectedhomeip/tree/master/src/controller/python)
+    or
     [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
     or
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
+    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin/CHIPTool)
+    or
+    [Android chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool)
     to communicate with the device.
 
 Note: The ESP32 does not support 5GHz networks. Also, the Device will persist
 your network configuration. To erase it, simply run.
 
           $ idf.py erase_flash ESPPORT=/dev/tty.SLAB_USBtoUART
 
-### Use the ESP32's Network
+### Setting up python controller
 
-Alternatively, you can connect to the ESP32's Soft-AP directly.
+Once ESP32 is up and running, we need to setup device controller to perform
+commissioning and cluster control.
 
-1.  After the application has been flashed, connect to the ESP32's Soft-AP. If
-    you use the M5Stack, the Soft-AP's SSID is encoded in the TLV section of the
-    QRCode on screen. It's usually something like `CHIP-XXX` where the last 3
-    digits are from the setup payload discriminator.
+-   Set up python controller
 
-2.  Once you're connected, the server's IP can be found at the gateway address.
+           $ cd {path-to-connectedhomeip}
+           $ ./scripts/build_python.sh -m platform
 
-3.  Use
-    [standalone chip-tool](https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool)
-    or
-    [iOS chip-tool app](https://github.com/project-chip/connectedhomeip/tree/master/src/darwin)
-    to communicate with the device.
+-   Execute the controller.
+
+           $ source ./out/python_env/bin/activate
+           $ sudo chip-device-ctrl
```

Comment:
This sudo bit is ... not desirable.  Why are we asking people to do that?

Reply:
Removed.